sympy/functions/elementary/exponential.py

```diff
@@ -975,17 +975,22 @@ def _eval_nseries(self, x, n, logx, cdir=0):
         #      IMPORTANT.
         from sympy.series.order import Order
         from sympy.simplify.simplify import logcombine
+        from sympy.core.symbol import Dummy
         _logx = logx
         if not logx:
             logx = log(x)
         if self.args[0] == x:
             return logx
         arg = self.args[0]
+        t = Dummy('t', positive=True)
+        if cdir == 0:
+            cdir = 1
+        z = arg.subs(x, cdir*t)
         k, l = Wild("k"), Wild("l")
-        r = arg.match(k*x**l)
+        r = z.match(k*t**l)
         if r is not None:
             k, l = r[k], r[l]
-            if l != 0 and not l.has(x) and not k.has(x):
+            if l != 0 and not l.has(t) and not k.has(t):
                 r = log(k) + l*logx  # XXX true regardless of assumptions?
```

Comment:
If cdir != 1, then `log(k*t**l)` will not be `log(k) + l*logx` as `log(x) = log(cdir) + log(t)`.

Reply:
Yeah , I think we should use `r = log(k) - l*log(cdir) + l*logx` rather !

Reply:
The simplest case which this block would  be addressing is `nseries` for `log(1/x)` and extrapolating the half ray logic( `x = cdir*t` where t is positive) that we used for leading term improves basic series result too .
```
>>> log(1/x)._eval_nseries(x, 4, logx=-p, cdir=1)
p
>>> log(1/x)._eval_nseries(x, 4, logx=-p, cdir=-1)     # on master p
p + 2*I*pi
```

Reply:
It seems that `match` does not make use of `t` being positive. The original code with `r = arg.match(k*x**l)` can be used and there is no need to even define the dummy `t` yet. The substitution `z = arg.subs(x, cdir*t)` could be done after this block.

Reply:
I guess we can use the dummy for the match block too , otherwise we won't necessarily have correct results for the basic cases
```
>>> log(1/x)._eval_nseries(x, 4, logx=-p, cdir=1)
p
>>> log(1/x)._eval_nseries(x, 4, logx=-p, cdir=-1)     # expected p + 2*I*pi
p
>>> log(-x)._eval_nseries(x, 4, logx=-p, cdir=1)
-p + I*pi
>>> log(-x)._eval_nseries(x, 4, logx=-p, cdir=-1)  # expected  -p - I*pi
-p + I*pi
```
Times when `arg` would be of the form `k*x**l`, we might not necessarily expect a series expansion out of it , so atleast we can comply to what limits/leading terms say in that case 
```
>>> log(1/x).as_leading_term(x, cdir=1)
-log(x)
>>> log(1/x).as_leading_term(x, cdir=-1)
-log(x) + 2*I*pi
```

Reply:
Itt seems that `- l*log(cdir)` will be needed in addition to `l*logx`. Could that be included in the definition of `logx`?

Reply:
Yes a log(cdir) will be involved here and I've made a commit too adding it according to our conversation here . I like the suggestion of appending `cdir` with `logx` . Will look into it and see how I can make that work .

Reply:
Maybe the name could be `logt` instead of `logx`. Then there would probably be no need for `_logx`.